src/frontend/core/executor/impl/PagerankExecutor.cpp

```diff
@@ -0,0 +1,325 @@
+                                          graphId.c_str(), PAGE_RANK, Conts::SLA_CATEGORY::LATENCY, partitionCount,
+                                          masterIP, autoCalibrate));
+        isStatCollect = true;
+    }
+
+    for (auto &&futureCall : intermRes) {
+        futureCall.get();
+    }
+
+    pageRank_logger.info(
+                "###PAGERANK-EXECUTOR### Getting Pagerank : Completed");
+
+    workerResponded = true;
+    JobResponse jobResponse;
+    jobResponse.setJobId(request.getJobId());
+    responseVector.push_back(jobResponse);
+
+    responseVectorMutex.lock();
+    responseMap[request.getJobId()] = jobResponse;
+    responseVectorMutex.unlock();
+
+    auto end = chrono::high_resolution_clock::now();
+    auto dur = end - begin;
+    auto msDuration = std::chrono::duration_cast<std::chrono::milliseconds>(dur).count();
+
+    std::string durationString = std::to_string(msDuration);
+
+    if (canCalibrate || autoCalibrate) {
+        Utils::updateSLAInformation(perfDB, graphId, partitionCount, msDuration, PAGE_RANK,
+                                    Conts::SLA_CATEGORY::LATENCY);
+        isStatCollect = false;
+    }
+
+    processStatusMutex.lock();
+    std::set<ProcessInfo>::iterator processCompleteIterator;
+    for (processCompleteIterator = processData.begin(); processCompleteIterator != processData.end();
+         ++processCompleteIterator) {
+        ProcessInfo processInformation = *processCompleteIterator;
+
+        if (processInformation.id == uniqueId) {
+            processData.erase(processInformation);
+            break;
+        }
+    }
+    processStatusMutex.unlock();
+}
+
+int PagerankExecutor::getUid() {
+    static std::atomic<std::uint32_t> uid{0};
+    return ++uid;
+}
+
+void PagerankExecutor::doPageRank(std::string graphID, double alpha, int iterations, string partition,
+                                 string host, int port, int dataPort, std::string workerList) {
+        if (host.find('@') != std::string::npos) {
+            host = Utils::split(host, '@')[1];
+        }
+
+        int sockfd;
+        char data[DATA_BUFFER_SIZE];
+        struct sockaddr_in serv_addr;
+        struct hostent *server;
+
+        sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+        if (sockfd < 0) {
+            std::cout << "Cannot create socket" << std::endl;
+            return;
+        }
+        server = gethostbyname(host.c_str());
+        if (server == NULL) {
+            pageRank_logger.error("ERROR, no host named " + host);
+            return;
+        }
+
+        bzero((char *)&serv_addr, sizeof(serv_addr));
+        serv_addr.sin_family = AF_INET;
+        bcopy((char *)server->h_addr, (char *)&serv_addr.sin_addr.s_addr, server->h_length);
+        serv_addr.sin_port = htons(port);
+        if (Utils::connect_wrapper(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
+            std::cout << "ERROR connecting" << std::endl;
```

Comment:
We should log this statement properly rather than output to console.

Reply:
logged the error

`pageRank_logger.error("Error connecting to socket");`

Reply:
Ok